native/src/rendering.cpp

```diff
@@ -431,6 +434,12 @@ void renderText(MapDataObject* obj, RenderableObject* rObj, RenderingRuleSearchR
 					if (rObj != NULL) {
 						renderableObjects.insert({info->object.id, rObj});
 					}
+
+					objPrimitive->primitiveText = info;
+
+					if (objPrimitive->prevObj && objPrimitive->prevObj->primitiveText) {
+						objPrimitive->prevObj->primitiveText->additionalIcon = ico;
```

Comment:
why do we store MapDataObjectPrimitive link, if only thing that's needed is SHARED_PTR<TextDrawInfo> ?

Reply:
Fixed